server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1157,29 +1163,39 @@ private void tryToUpdateInsertLastCache(InsertRowPlan plan, Long latestFlushedTi
       return;
     }
     IMeasurementMNode[] mNodes = plan.getMeasurementMNodes();
-    int columnIndex = 0;
-    for (IMeasurementMNode mNode : mNodes) {
-      // Don't update cached last value for vector type
-      if (!plan.isAligned()) {
-        if (plan.getValues()[columnIndex] == null) {
-          columnIndex++;
+    if (plan.isAligned()) {
+      for (int i = 0; i < mNodes.length; i++) {
+        if (plan.getValues()[i] == null) {
           continue;
         }
         // Update cached last value with high priority
-        if (mNode != null) {
+        // vector lastCache update need subSensor path
+        IoTDB.metaManager.updateLastCache(
+            plan.getPrefixPath().concatNode(plan.getMeasurements()[i]),
+            plan.composeTimeValuePair(i),
+            true,
+            latestFlushedTime,
+            mNodes[i]);
+      }
+    } else {
+      for (int i = 0; i < mNodes.length; i++) {
```

Comment:
Should we put the `if (plan.isAligned)` in the for loop to reduce the line number?